### PR TITLE
feat: add platform ssh docs

### PR DESCRIPTION
**Pull request by beguene**
This is a draft version of the documentation for the platform ssh or how to setup a local editor for the Hiro Platform.
Not to be merged yet as the feature is not yet released on the platform.

### CODE REVIEW

File: docs/platform/setting-up-local-editor.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+---
+title: Setting Up Your Local Editor
+---
+# Setting Up Your Local Editor
+
+Hiro Platform now supports local development, allowing you to use your preferred local editor (VSCode, Emacs, Vim, etc.) to develop smart contracts. This guide will walk you through the steps to set up your local editor for Hiro Platform.
+
+## Step 1: Find or Generate your SSH Key
+
+First, you need to check if you already have an existing SSH key on your device. This is often located in `~/.ssh/id_ed25519.pub`. You can view your SSH keys by running the following command in your terminal:
```
Comment (sabbyanandan) [inline docs/platform/setting-up-local-editor.md:10]: To be consistent with the rest of the Platform guides, perhaps we can remove the `Step X`?

**Review by sabbyanandan — CHANGES_REQUESTED** [review]
This is looking great! I left a few suggestions. We could also add the screenshots when they are ready.

**Comment by andresgalante** [comment]
@mefrem would you be able to review this one, please?

File: docs/platform/connect-desktop-to-cloud.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+```bash
+ssh-keygen -t ed25519
+```
+
+You'll then be prompted to enter a passphrase. You can choose to leave that blank. You should now have a `id_ed25519.pub` file which contains your new public SSH key.
+
+## Save your SSH Key
+
+Next, you'll provide your public key to the Hiro Platform. You can do this by copying your public key and pasting it into the SSH Public Key field in the Hiro Platform.
```
Comment (ryanwaits) [inline docs/platform/connect-desktop-to-cloud.md:28]: might be nice to add a code snippet for how to copy pk, ex:

for mac: ```cat ~/.ssh/id_rsa.pub | pbcopy```

also, where in the platform would someone find where to paste in the public key? i did a quick check and wasn't sure where to add it.

File: docs/platform/connect-desktop-to-cloud.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+To run authenticated Git operations from within your Hiro Platform workspace, you'll need to enable SSH port forwarding. If you don't plan to use Git, you can skip this step.
+
+Open the file at `~/.ssh/config` with a text editor, or create it if it doesn't exist. Enter the following text into the file:
+
+```bash
+Host <your-workspace-url> 
+    ForwardAgent yes
+```
+
+Replace `<your-workspace-url>` with the actual URL of your pod, which should look something like this: https://4a1b57d0c473.platform.hiro.so/
```
Comment (ryanwaits) [inline docs/platform/connect-desktop-to-cloud.md:43]: maybe add some context on how to find the unique pod url, ie you must open a specific project in the editor to find it (unless theres somewhere else in the app where you can find)

File: docs/platform/connect-desktop-to-cloud.md (hunk 2 above)
Comment (beguene) [reply docs/platform/connect-desktop-to-cloud.md]: I added for Linux, Mac and Windows and a short phrase and a direct link to the settings (where the user can add the ssh keys)

**Comment by beguene** [comment]
@andresgalante @ryanwaits Shall I merge this ?

**Comment by beguene** [comment]
@sabbyanandan I made the changes you requested, but I can't merge it until you approve. Let me know if there are still things missing.

File: docs/platform/connect-desktop-to-cloud.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+```
+cat ~/.ssh/id_ed25519.pub | xclip -selection clipboard
+
+```
+
+You'll also need to provide a nickname for your connected device, such as 'MacBook Pro'. This will help you identify the device in the future.
+
+## Enable SSH and Git
+
+To enable SSH access and Git authentication to your Hiro Platform workspace, you'll need to add the following lines to your `~/.ssh/config` file:
```
Comment (sabbyanandan) [inline docs/platform/connect-desktop-to-cloud.md:54]: Aside from the instructions, do we want to explain what this does? This is so that users are aware of copying this command into their machine (with root access).

File: docs/platform/connect-desktop-to-cloud.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+If you don't plan to use Git, you can delete the line 
+
+```bash
+    ForwardAgent yes
+```
+
+## Get your workspace url
+
+<!-- markdown-link-check-disable -->
+To create an SSH connection to your Hiro Platform workspace you will need your workspace url. To find it just go to [https://platform.hiro.so/projects](https://platform.hiro.so/projects) and click on `Connect Your Local Editor`. You will find your worspace url at the bottom.
```
Comment (sabbyanandan) [inline docs/platform/connect-desktop-to-cloud.md:72]:
```suggestion
To create an SSH connection to your Hiro Platform workspace you will need your workspace url. To find it just go to [https://platform.hiro.so/projects](https://platform.hiro.so/projects) and click on `Connect Your Local Editor`. You will find your workspace url at the bottom.
```

File: docs/platform/connect-desktop-to-cloud.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+<!-- markdown-link-check-enable -->
+
+## Connection Complete!
+
+Now you can access your Hiro Platform projects directly from your local setup. Use an SSH client or editor with SSH support to open your projects. 
+
+## Configuring Visual Studio Code
+
+Visual Studio Code (VSCode) has built-in support for SSH connections, allowing you to connect to and work on your remote server directly from your local VSCode environment. Here's how to set it up:
```
Comment (sabbyanandan) [inline docs/platform/connect-desktop-to-cloud.md:81]: The user still has to install "Remote-SSH" extension if they don't have it already, though. So I would rephrase this language.

File: docs/platform/connect-desktop-to-cloud.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+1. Open VSCode on your local machine.
+
+2. Open the Command Palette by pressing `F1` or `Ctrl+Shift+P` (or `Cmd+Shift+P` on macOS).
+
+3. Type "Remote-SSH: Connect to Host..." and select it from the dropdown list.
```
Comment (sabbyanandan) [inline docs/platform/connect-desktop-to-cloud.md:87]: If this extension is not available already, user must install it.

**Review by sabbyanandan — CHANGES_REQUESTED** [review]

**Review by sabbyanandan — APPROVED** [review]
Let's go!
